samples/fileSetMetadata.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+// sample-metadata:
+//   title: Storage Set File Metadata.
+//   description: Set file metadata.
+//   usage: node fileSetMetadata.js <BUCKET_NAME> <FILE_NAME>
+
+async function main(bucketName = 'my-bucket', filename = 'file.txt') {
+  // [START storage_set_metadata]
+  // Imports the Google Cloud client library
+  const {Storage} = require('@google-cloud/storage');
+
+  // Creates a client
+  const storage = new Storage();
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const bucketName = 'Name of a bucket, e.g. my-bucket';
+  // const filename = 'File to access, e.g. file.txt';
+
+  const [metadata] = await storage
```

Comment:
If a user copies and pastes the sample between the region tags here, their code won't run because of the top level `await`.  We should wrap this specific piece of code in side of an async function, then call it inside the region tag:

```js
async function setFileMetadata() {
  const [metadata] = await storage....
  ....
}
setFileMetadata();
```

I know this smells, but it's pretty frequent feedback and part of the style guide.  Adding @bcoe in case he has an idea of how to catch this with a lint rule or somethin'

Reply:
it's hard, because it's not _technically_ a top level `await` which I'm sure linting could catch, it's just a top-level await in the context of the region tags.

So, for what it's worth, I saw that top-level await just landed in V8, so I wouldn't be shocked if we have it upstreamed to Node 12/13 ... the wrapping function might become less necessary over the next year.